--- alsaconf.old	2011-02-15 14:03:32.756666991 -0200
+++ alsaconf	2011-02-15 14:07:10.410000330 -0200
@@ -55,6 +55,8 @@
   suse_version=$(grep 'VERSION = ' /etc/SuSE-release | sed -e s/'VERSION = '//)
 elif [ -f /etc/UnitedLinux-release ]; then
   distribution="suse"
+elif test -f /etc/arch-release && grep -q "Arch" /etc/arch-release; then
+  distribution="arch"
 elif [ -f /etc/gentoo-release ]; then
   distribution="gentoo"
 elif [ -f /etc/debian_version ]; then
@@ -291,7 +293,7 @@
 
 # cfgfile = base config file to remove/update the sound setting
 # cfgout = new config file to write the sound setting (if different from $cfgfile)
-cfgout=""
+cfgout="/etc/modprobe.d/sound.conf"
 cfgoldout=""
 if [ -n "$cfgfile" ]; then
   if [ ! -r "$cfgfile" ]; then
@@ -508,6 +510,8 @@
     rcalsasound=/etc/init.d/alsa
 elif [ -x /etc/rc.d/rc.alsa ]; then
     rcalsasound=/etc/rc.d/rc.alsa
+elif [ -x /etc/rc.d/alsa ]; then
+	rcalsasound=/etc/rc.d/alsa
 elif [ -x /etc/init.d/alsasound ]; then
     rcalsasound=/etc/init.d/alsasound
 elif [ -x /usr/sbin/rcalsasound ]; then
